### DISCUSSION TITLE
Pore in multiscale tesselation

### DISCUSSION THREAD

**Discussion by MaximePrs** [neperfepx/neper#765] [Q&A] [answered]
Hello everyone,

As discussed in #764, I am having a problem generating a typical microstructure.

For more information, using the command line:
`neper -T -domain 'square(74,74)' -n '100::from_morpho' -dim 2 -morpho "diameq:lognormal(2.40,1.17)::diameq:0.5*lognormal(1.22,0.2)+0.5*lognormal(1.22,0.2),aspratio(4,1)" -group mode -statcell id,group,"scaleid(1),scaleid(2)",diameq,area,circularity -o morpho_1`

I get a 2-scale microstructure with 2 phases (phase 1 in red and phase 2 in green):
![M_1100_m-ech_taille-DTTH_asp-4-1_id1_2](https://github.com/neperfepx/neper/assets/111365721/6449dd72-bd66-486f-bac5-b0875e943431)

Thanks to the scaleid(1) information, I can kind of "merge" the grains of the phase 2 with the same scaleid(1) by giving them the same crystal orientation (doing it with Python code for another software).
For example, here are the first lines of the .stcell file:
```
id group scaleid(1) scaleid(2) diameq area circularity
1 1 1 1 1.179406936115 1.092489411523 0.626321626856
2 2 1 2 1.492027700138 1.748411496631 0.896896712651 --> crystal orientation 1
3 1 2 1 1.190530138592 1.113193520227 0.763917700777
4 1 2 2 1.079371795216 0.915023003429 0.632830297648
5 2 2 3 1.190321452619 1.112803294889 0.851350219472 --> crystal orientation 2
6 2 2 4 1.564533887444 1.922471144633 0.771968031299 --> crystal orientation 2
7 2 2 5 1.307464446951 1.342609320539 0.665653168565 --> crystal orientation 2
8 1 2 6 0.901949740622 0.638931878900 0.546664410509
9 1 2 7 1.364905856090 1.463171642517 0.645084789806
10 2 2 8 1.052142542262 0.869438852897 0.799506304118 --> crystal orientation 2
```
If the crystal orientation is characterized by a color, it gives this picture for the grains of the phase 2:
![M_1100_m-ech_taille-DTTH_asp-4-1_id1_4](https://github.com/neperfepx/neper/assets/111365721/121e7f52-075d-4828-955b-eb48f818c0f4)

So far there is no problem.
But as said in #764 
I'm interested in adding a pore near the surface (see picture below) while keeping all the scaleid information to "merge" the grains of the phase 2 manually with python.  I tried with `-transform "cut(cylinder())"` but I can't keep the scaleid information.
![Por_neper_2](https://github.com/neperfepx/neper/assets/111365721/02f31789-0456-48e7-bed2-646fdab5bf0c)

As I'm not an expert in neper, I'm looking for help from the community:
Is there another way to realize what I'm doing ?
Or another solution to add a pore to my microstructure ?

Best,

**Comment by rquey** [comment, marked as answer]
This discussion has been closed as obsolete. Please repost if necessary.
